manager/controlapi/node.go

```diff
@@ -13,7 +13,30 @@ func validateNodeSpec(spec *api.NodeSpec) error {
 	if spec == nil {
 		return grpc.Errorf(codes.InvalidArgument, errInvalidArgument.Error())
 	}
-	return nil
+
+	var me MultiError
+
+	if err := validateAnnotations(spec.Annotations); err != nil {
+		me = append(me, err)
+	}
+
+	if spec.Membership != api.NodeMembershipPending && spec.Membership != api.NodeMembershipAccepted {
+		err := grpc.Errorf(codes.InvalidArgument, "NodeSpec: membership should be in range 0-1")
+		me = append(me, err)
+	}
+
+	if spec.Availability != api.NodeAvailabilityActive &&
+		spec.Availability != api.NodeAvailabilityPause &&
+		spec.Availability != api.NodeAvailabilityDrain {
+		err := grpc.Errorf(codes.InvalidArgument, "NodeSpec: Availability should be in range 0-2")
```

Comment:
I'd prefer not to mention protobuf field numbers in error messages. They won't be meaningful to the user.